appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy

```diff
@@ -112,6 +114,122 @@ class CreateUserOnBuddyRequestTest extends AbstractAppServiceIntegrationTest
 		appService.deleteUser(updatedBob)
 	}
 
+	def 'Bob adjusts data, including his Android device'()
+	{
+		given:
+		def richard = addRichard()
+		def mobileNumberBob = makeMobileNumber(timestamp)
+		def inviteUrl = buildInviteUrl(sendBuddyRequestForBob(richard, mobileNumberBob))
+		def bob = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateDataCreatedOnBuddyRequest, inviteUrl, true, null)
+
+		when:
+		def newNickname = "Bobby"
+		def updatedBobJson = bob.convertToJson()
+		updatedBobJson.nickname = newNickname
+		User bobToBeUpdated = new User(updatedBobJson)
+		bobToBeUpdated.deviceName = "My S8"
+		bobToBeUpdated.deviceOperatingSystem = "ANDROID"
+		User updatedBob = appService.updateUserCreatedOnBuddyRequest(appService.&assertUserUpdateResponseDetails, bobToBeUpdated, inviteUrl)
+
+		then:
+		updatedBob.devices.size() == 0 // Mobile number not confirmed yet
```

Comment:
Wouldn't it be more logical to return the devices here, even if the mobile number is not confirmed yet? Because the user is also allowed to save them.

Reply:
Not really. They can create the user entity with a device but can't do anything with devices. We are till now leaving out everything till the number is confirmed. I don't think there would be any benefit if we would make an exception for the device.